presto-spark-base/src/test/java/com/facebook/presto/spark/TestPrestoSparkQueryRunner.java

```diff
@@ -1340,6 +1341,103 @@ public void testCreatePartitionedTable()
         assertQuerySucceeds("DROP TABLE test_partition_table");
     }
 
+    @Test
+    public void testDisableBroadcastJoinExecutionStrategy()
+    {
+        Session session = Session.builder(getSession())
+                .setSystemProperty(JOIN_DISTRIBUTION_TYPE, "BROADCAST")
+                .setSystemProperty(STORAGE_BASED_BROADCAST_JOIN_ENABLED, "true")
+                .setSystemProperty(SPARK_BROADCAST_JOIN_MAX_MEMORY_OVERRIDE, "10B")
+                .setSystemProperty(SPARK_RETRY_ON_OUT_OF_MEMORY_BROADCAST_JOIN_ENABLED, "false")
+                .build();
+
+        // Query should fail with broadcast join OOM
+        assertQueryFails(
+                session,
+                "select * from lineitem l join orders o on l.orderkey = o.orderkey",
+                "Query exceeded per-node broadcast memory limit of 10B \\[Broadcast size: .*MB\\]");
+
+        session = Session.builder(getSession())
+                .setSystemProperty(JOIN_DISTRIBUTION_TYPE, "BROADCAST")
+                .setSystemProperty(STORAGE_BASED_BROADCAST_JOIN_ENABLED, "true")
+                .setSystemProperty(SPARK_BROADCAST_JOIN_MAX_MEMORY_OVERRIDE, "10B")
+                .setSystemProperty(SPARK_RETRY_ON_OUT_OF_MEMORY_BROADCAST_JOIN_ENABLED, "false")
+                .setSystemProperty(SPARK_EXECUTION_STRATEGIES, "DISABLE_BROADCAST_JOIN")
+                .build();
+
+        // Query should succeed since broadcast join will be disabled due to present of DISABLE_BROADCAST_JOIN tag
```

Comment:
Wondering if we should update this comment and similarly comments in below test cases to callout that we are passing SPARK_QUERY_EXECUTION_STRATEGIES explicitly and disabling the retry. The query should succeed since DISABLE_BROADCAST_JOIN strategy will be applied in the first run itself.